### ISSUE TITLE
[Bug] LinkTo `@query` assertion regression

### ISSUE THREAD

**Issue by Windvis** [emberjs/ember.js#19693]
### 🐞 Describe the Bug
The 3.27 release includes the modernized built-in components feature. The new code also adds an [extra assertion](https://github.com/emberjs/ember.js/commit/3d027841b30660dbaf136b0590f0c09a6e1317ce#diff-4ef2173974821b22319e3588d4b1fbba945c84e9997aad54e722787802fbdaa9R233-R236) for the `@query` argument. This assertion however is more strict than before. Passing undefined causes it to throw, which wasn't the case before. This means that code depending on this not throwing will fail in 3.27+, which is a breaking change.

**Edit** You can work around it by passing an empty object instead of undefined if no query params need to be passed.

This is especially tricky when creating wrapper components which need to support the same arguments as  `<LinkTo>`. Since Ember doesn't support "spreading" arguments, passing through undefined when arguments don't have a value is one of the only DX friendly alternatives.

Wrapping `<LinkTo>` was already tricky since `@model` and `@models` have similar assertions (which were also added during the 3.x lifecycle) but you can also work around them by converting `@model` to an array and passing it as `@models` instead.

### 🔬 Minimal Reproduction
Create a component which wraps `<LinkTo>` and directly passes through a `@query` argument. Invoke the wrapper component without passing a value to the `@query` argument.

```hbs
{{! project-link.hbs}}
<LinkTo
  @route={{@route}}
  @models={{some-helper-with-similar-logic-as-link-to @model @models}}
  @query={{@query}}
>
  {{yield}}
</LinkTo>
```

```hbs
<ProjectLink @route="route-name">Link</ProjectLink>
```
I can create a reproduction or failing test if needed.

### 😕 Actual Behavior
An assertion is thrown if `undefined` is passed into the `@query` argument.

### 🤔 Expected Behavior
No assertion is thrown. `undefined` is handled the same as if no value was passed.

### 🌍 Environment

- Ember: 3.27
- Node.js/npm: 12 / 6
- OS: Linux
- Browser: Firefox

### ➕ Additional Context
Discord discussion: https://discord.com/channels/480462759797063690/485447409296736276/870747627040436325

**Comment by Techn1x** [comment]
also experiencing this, btw

**Comment by panthony** [comment]
Updated to ember 3.28 and I just stumbled upon this assertion for all the LinkTo, even as simple as:

```hbs
<LinkTo @route="routeName">Label</LinkTo>
```

Something seems off, can't be legit to have this failing.

**Comment by Windvis** [comment]
@panthony Are you extending or overriding the LinkTo component in some way? This assertion should only be thrown if you pass in `undefined` into the `@query` argument, which you don't seem to be doing here?

**Comment by Windvis** [comment]
FYI, I closed this issue because there was no core-member response to this issue at all and Ember 4.x has since been released. Even if it is considered a bug I don't see it being fixed and backported to the 3.28 branch with the current development activity and 3.28 only receiving bugfixes until August 29, 2022. So I just assumed this is a "will not fix" issue.

**Comment by panthony** [comment]
@Windvis I'm not extending it, only wrapping like you did with `ProjectLink` where it made sense.

The snippet I gave is legit something that fails (minus the actual label/route name).

I'll try to dig deeper because that is really weird.

If I simply bump to ember 3.28 I do not have this issue, but it appeared when I started updating dependencies that triggered the babel deprecation warning (ember-promise-helpers, ember-test-selectors, ember-truth-helpers, ember-copy, ..).

--

Thank you about the explanation about why the issue was closed without obvious resolution and/or response.

I won't take more of your time, worst case I'll wrap the component in all cases, the amount of occurrences is still manageable.
